src/data/list/basic.lean

```diff
@@ -3222,6 +3243,46 @@ theorem mem_bag_inter {a : α} : ∀ {l₁ l₂ : list α}, a ∈ l₁.bag_inter
       rintro ⟨rfl, h'⟩, exact h.elim h' }
   end
 
+@[simp] theorem count_bag_inter {a : α} :
+  ∀ {l₁ l₂ : list α}, count a (l₁.bag_inter l₂) = min (count a l₁) (count a l₂)
+| []         l₂ := by simp
+| l₁         [] := by simp
+| (h₁ :: l₁) (h₂ :: l₂) :=
+begin
+  dsimp [list.bag_inter],
```

Comment:
I was in the middle of cleaning up this proof when I noticed @cipher1024 had already merged. It still strikes me as much longer than necessary. I've shaved off ten lines just by combining some `rw`, `simp`, `conv`, `subst` successions, and I feel like it could be a lot shorter.

I only had a few minutes to try this and won't be able to keep working tonight, but I'll push the current status to the `bag_inter` branch.

Reply:
Thanks @robertylewis for cleaning this up a bit. Sorry I wrote such a lame proof. :-) I have trouble getting excited about golfing content-free proofs (surely one day this proof will disappear, sinking below the rising tide of automation...), but appreciate that I should try to do better for mathlib PRs.

Reply:
I look at things differently. Since this is a content-free proof, the fact that it's 30 lines long is unacceptable. The proof is entirely by case splits and rewriting, so it should follow using simp at most once per case. If it takes some crazy chain of backwards rewrites, convs, and simp onlys, then either there are missing simp lemmas or something is wrong with the approach.

It's down to 10 lines and I'm running out of patience; when Travis finishes on the `bag_inter` branch I'll merge.